app/models/project.rb

```diff
@@ -22,6 +22,9 @@ class Project < ApplicationRecord
   TYPE_TASK = { project_type: 0, test_type: 1 }.freeze
   enum project_type: TYPE_TASK
 
+  STATUSES = { active: 0, notification_sent: 1, completed: 2, favorite: 3 }.freeze
```

Comment:
i suggest: STATUSES = %i[active notification_sent completed favorite].freeze

Reply:
This approach is not at all flexible. Imagine that the requirements just changed. From now on, "message_sent" should be divided into "client" and "administrator". In this case, you must delete the old value and add two new ones. But ... you cannot delete an unused "alien" type because it violates the order of the remaining values. To avoid this situation, enum was declared as Hash.

---
sources:
https://naturaily.com/blog/ruby-on-rails-enum
https://api.rubyonrails.org/v5.2.3/classes/ActiveRecord/Enum.html